src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -943,57 +972,75 @@ internal virtual void PrepareSession()
+                WebSession.Handler = handler = new();
 
-            if (WebSession.Certificates is not null)
-            {
-                handler.ClientCertificates.AddRange(WebSession.Certificates);
-            }
+                handler.CookieContainer = WebSession.Cookies;
+                handler.AutomaticDecompression = DecompressionMethods.All;
 
-            if (SkipCertificateCheck)
-            {
-                handler.ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
-                handler.ClientCertificateOptions = ClientCertificateOption.Manual;
-            }
+                // Set the credentials used by this request
+                if (WebSession.UseDefaultCredentials)
+                {
+                    // The UseDefaultCredentials flag overrides other supplied credentials
+                    handler.UseDefaultCredentials = true;
+                }
+                else if (WebSession.Credentials is not null)
+                {
+                    handler.Credentials = WebSession.Credentials;
+                }
 
-            // This indicates GetResponse will handle redirects.
-            if (handleRedirect || WebSession.MaximumRedirection == 0)
-            {
-                handler.AllowAutoRedirect = false;
-            }
-            else if (WebSession.MaximumRedirection > 0)
-            {
-                handler.MaxAutomaticRedirections = WebSession.MaximumRedirection;
-            }
+                if (WebSession.NoProxy)
+                {
+                    handler.UseProxy = false;
+                }
+                else if (WebSession.Proxy is not null)
+                {
+                    handler.Proxy = WebSession.Proxy;
+                }
 
-            handler.SslProtocols = (SslProtocols)SslProtocol;
+                if (WebSession.Certificates is not null)
+                {
+                    handler.ClientCertificates.AddRange(WebSession.Certificates);
+                }
 
-            HttpClient httpClient = new(handler);
+                if (WebSession.SkipCertificateCheck)
+                {
+                    handler.ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
+                    handler.ClientCertificateOptions = ClientCertificateOption.Manual;
+                }
+                handler.AllowAutoRedirect = WebSession.AllowAutoRedirect;
+                if (WebSession.AllowAutoRedirect && WebSession.MaximumRedirection > 0)
+                {
+                    handler.MaxAutomaticRedirections = WebSession.MaxAutomaticRedirections;
+                }
+                handler.SslProtocols = (SslProtocols)WebSession.SslProtocol;
+            }
+            // Only dispose the httpClientHandler if we are not trying to use a persistent one
+            HttpClient httpClient = new(handler, _noWebSession && string.IsNullOrEmpty(SessionVariable));
```

Comment:
Maybe you can replace _noWebSession with a helper method _noWebSession => MyInvocation.BoundParameters.ContainsKey("WebSessionl"))